spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -668,5 +667,27 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
               ): Array[(ValidationResult, ValidationMethod[T])] = {
     Validator(this, dataSet).test(vMethods)
   }
+
+
+  /**
+   * Generate end nodes of current module with start nodes
+   * @param startNodes: current start nodes
+   * @return current end nodes
+   */
+  def getEndNodes(startNodes: Array[ModuleNode[T]]): Array[ModuleNode[T]] = {
+    val endNodes = Array(this.inputs(startNodes: _*))
+    endNodes
+  }
+
+  def toGraph(): Module[T] = {
+    val startNodes = Array(Input[T]())
+    toGraph(startNodes)
+  }
+
+  def toGraph(startNodes: Array[ModuleNode[T]]): Module[T] = {
```

Comment:
Add Java doc. Change return type from Module[T] to Graph[T] and change Array to Seq